intake_netcdf/__init__.py

```diff
@@ -70,17 +77,17 @@ def _get_schema(self):
 
     def read(self):
         self._load_metadata()
-        return self._ds
+        return self._ds.load()
 
     def read_chunked(self):
-        raise Exception('read_chunked not supported for xarray containers.')
+        self._load_metadata()
+        return self._ds
 
     def read_partition(self, i):
         raise Exception('read_partition not supported for xarray containers.')
```

Comment:
@martindurant read_partition is not implemented yet. I'm wondering if we need to support it.

Reply:
Right. To my mind, it should be implemented at some point, but the input is a tuple. I would keep an exception for now (but should be NotImplemented, because we may do it later).